packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/NamedTypeSymbolProviders/NamedTypeSymbolProviderTests.cs

```diff
@@ -109,7 +114,9 @@ public void ValidatePropertyTypes(Type propertyType)
             Assert.AreEqual(expectedType.IsList, propertyCSharpType.IsList);
             Assert.AreEqual(expectedType.Arguments.Count, propertyCSharpType.Arguments.Count);
             Assert.AreEqual(expectedType.IsCollection, propertyCSharpType.IsCollection);
-            Assert.AreEqual(expectedType.IsFrameworkType, propertyCSharpType.IsFrameworkType);
+
+            var isFrameworkType = !isEnum && expectedType.IsFrameworkType;
```

Comment:
Why do we need the isEnum check?

Reply:
For the test cases that are nullable types, we are using this ctor for constructing the expected CSharpType `new CSharpType(propertyType, isNullable)` which will always set a FrameworkType. That isn't always accurate since for non-framework types, we are constructing the CSharpType from the type symbol manually and the framework type property will always be null.

Reply:
So can we instead update the logic that creates the CSharpType to not use the Type based ctor for System.Nullable types? I think this should match what we do in NamedTypeSymbolProvider.

Reply:
sure, I've updated the test to more closely reflect the expected behavior.